shared/src/main/scala/amf/core/internal/plugins/syntax/StringDocBuilder.scala

```diff
@@ -0,0 +1,144 @@
+package amf.core.internal.plugins.syntax
+
+import amf.core.client.common.position.Position
+import amf.core.client.common.position.Position._
+import amf.core.client.scala.parse.document.{ParsedDocument, StringParsedDocument}
+import amf.core.internal.plugins.syntax.StringDocBuilder.INDENTATION_WIDTH
+import org.yaml.builder.DocBuilder
+
+import scala.collection.mutable
+
+class SourceCodeBlock(val indentation: Integer, val lines: mutable.Buffer[(String, Position)] = mutable.Buffer()) {
+
+  def sortedBuilder: StringBuilder = {
+    val b            = new StringBuilder()
+    var previousLine = minPos.line
+    lines.sortBy(_._2).foreach {
+      case (l, p) =>
```

Comment:
Not related to this line. Might it be useful to separate builder factory functionality from this class? It seems cleaner as we separate creation functionality from "collection like" functionality

Reply:
Could be, that will enabled have the lines as an inmutable list as well. But I hope that we can talked that lather, when the real development of this feature starts.